src/main/scala/essential_scala/Ex.scala

```diff
@@ -0,0 +1,126 @@
+package essential_scala
+
+object Ex extends App {
+
+  // 2.2.5.1 OperatorStyle
+  "foo".take(1)
+  "foo" take 1
+    1.+(2).+(3)
+
+  // 2.3.8.1 LiterallyJustLiterals
+  // Int Boolean Long Double
+
+  // 2.3.8.2 Quotesand Misquotes
+  // Character String
+
+  // 2.3.8.3 AnAside on Side-Effects
+  // String Unit
+
+//  2.4.5.1 Cat_o_matique
+
+  object Cat1 {
+    val name = "Oswald"
+    val colour = "Black"
+    val food = "Milk"
+  }
+
+  object Cat2 {
+    val name = "Henderson"
+    val colour = "Ginger"
+    val food = "Chips"
+  }
+
+  object Cat3 {
+    val name = "Quentin"
+    val colour = "Tabby and white"
+    val food = "Curry"
+  }
+
+// 2.4.5.2 Square Dance!
+  object calc {
+    def square(l: Double): Double = l*l
+    def cube(l: Double): Double = l*square(l)
+  }
+
+// 2.4.5.4 Order of evaluation
+  // a = 1, b = 3, c = "3c"
+  // 3c31
+
+// 2.4.5.5 Gree?ngs,human
+  object person {
+    val firstName = ""
+    val lastName = ""
+  }
+
+  object alien {
+    def greet(p: person.type): String = "Greetings, " + p.firstName
+    // No
+  }
+
+//  3.1.6.1 Cats,Again
+  class Cat1(
+    val colour: String,
+    val food: String
+  )
+  val oswald = new Cat1("Black", "Milk")
+  val henderson = new Cat1("Ginger", "Chips")
+  val quentin = new Cat1("Tabby and white", "Curry")
+
+//  3.1.6.2 Catson the Prowl
+  object ChipShop {
+    def willServe(cat: Cat1) = cat.food == "Chips"
+  }
+//  3.4.5.1 Case Cats
+case class Cat2(
```

Comment:
Suggestion: use `scalafmt` sbt plugin for the project to avoid misformatting.

Reply:
Use scalafmt plugin.